### PR TITLE
Adopt voting flow to new design - Closes #4425

### LINKED ISSUE #4425: Adopt voting flow according to new design
### Description

- Implement new [design](https://www.figma.com/file/9JbvSmtLNJIttXp68uhmQR/Change-API-to-support-Lisk-SDK-v6-API?node-id=98%3A16082) changes for voting domain
  - Update delegates page to show 
    - Summary on hover (mini profile)
    - Allow to add votes
  - Update Add voting modal
  - Update Edit voting modal

### Acceptance Criteria

- Users should be able to add/edit vote for a delegate from delegate domain or mini profile
- User should be able to see warning for punished delegate
- Add/Update required unit tests

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4425

### How was it solved?

- [x] Refactor existing EditVote dialog to use required query hooks
- [x] Refactor UI to match the new designs
- [x] Implement the second modal step of the vote flow
- [x] Update unit tests

### How was it tested?

Jest
Visually

### CODE REVIEW

File: src/modules/dpos/validator/components/DelegateSummary/DelegateSummary.js [hunk 1]
```diff
@@ -33,16 +28,16 @@ const DelegateSummary = ({
...
+              </PrimaryButton>
+            </DialogLink>
           </div>
         </div>
       </BoxHeader>
       <BoxContent>
-        <p>
-          {t('This delegate is among the first 101 delegates in delegate weight ranking.')}
-        </p>
+        <p>{t('This delegate is among the first 101 delegates in delegate weight ranking.')}</p>
```
Comment (reyraa) [inline src/modules/dpos/validator/components/DelegateSummary/DelegateSummary.js:41]: 101 should not be hard-coded.

File: src/modules/dpos/validator/components/EditVote/index.js [hunk 2]
```diff
@@ -1,148 +1,235 @@
...
 
-  const confirm = () => {
+  const handleConfirm = () => {
+    if (!isForm) {
+      removeThenAppendSearchParamsToUrl(history, { modal: 'votingQueue' }, ['modal']);
+      return;
+    }
     voteEdited([
       {
-        address: address || wallet.summary.address,
```
Comment (reyraa) [inline src/modules/dpos/validator/components/EditVote/index.js:59]: This was used to handle other delegate profiles

File: src/modules/dpos/validator/components/WarnPunishedDelegate/index.js [hunk 3]
```diff
@@ -1,79 +1,59 @@
...
+//   currentHeight: state.blocks.latestBlocks.length ? state.blocks.latestBlocks[0].height : 0,
+// });
 
 const getPunishmentDetails = (punishedTimestamp, pomHeight, currentHeight) => {
   const startDate = new Date(punishedTimestamp * 1000);
   const punishmentStartDate = moment(startDate).format('MM.DD.YYYY');
-  // 6: blocks per minute, 60: minutes, 24: hours
-  const numOfBlockPerDay = 24 * 60 * 6;
-  const daysLeft = Math.ceil((pomHeight.end - currentHeight) / numOfBlockPerDay);
+  console.log('...', (pomHeight.end - currentHeight), currentHeight, pomHeight.end)
```
Comment (reyraa) [inline src/modules/dpos/validator/components/WarnPunishedDelegate/index.js]: Remove the log.

File: src/modules/dpos/validator/utils/getMaxAmount.js [hunk 4]
```diff
@@ -39,26 +49,36 @@ const getMaxAmount = async (wallet, network, voting, address) => {
...
-    network,
-    transaction,
-    selectedPriority: { title: 'Normal', value: 0, selectedIndex: 0 }, // Always set to LOW
-    numberOfSignatures: getNumberOfSignatures(wallet, transaction),
-  }, 'LSK');
+  const maxAmountFee = await getTransactionFee(
+    {
+      network,
+      transaction,
+      token: 'LSK',
```
Comment (reyraa) [inline src/modules/dpos/validator/utils/getMaxAmount.js:63]: Should this be hard-coded? can we read it from the store?

File: src/modules/dpos/validator/utils/getMaxAmount.js [hunk 5]
```diff
@@ -39,26 +49,36 @@ const getMaxAmount = async (wallet, network, voting, address) => {
...
+      wallet: {
+        summary: { isMultisignature },
+        keys: { members: [...optionalKeys, ...mandatoryKeys] },
+      },
+      selectedPriority: { title: 'Normal', value: 0, selectedIndex: 0 }, // Always set to LOW
```
Comment (reyraa) [inline src/modules/dpos/validator/utils/getMaxAmount.js:69]: Also here. I think this should be handled by the hook. By adding it here you are parallelizing the priority selection logic, which is anti-pattern.

File: src/modules/dpos/validator/utils/getMaxAmount.js [hunk 6]
```diff
@@ -39,26 +49,36 @@ const getMaxAmount = async (wallet, network, voting, address) => {
...
+      numberOfSignatures: getNumberOfSignatures(
+        { numberOfSignatures, isMultisignature },
+        transaction
+      ),
+    },
+    'LSK'
```
Comment (reyraa) [inline src/modules/dpos/validator/utils/getMaxAmount.js:76]: LSK must not be hard-cocded.

**Review by reyraa — CHANGES_REQUESTED** [review]

File: src/modules/dpos/validator/components/WarnPunishedDelegate/index.js [hunk 7]
```diff
@@ -1,79 +1,54 @@
...
-    }
-  }, [props.pomHeight.start]);
+  const { pomHeight } = props;
+  const { t } = useTranslation();
+
+  const { data: blocks } = useBlocks();
+  const { data: blocksAtHeight } = useBlocks({ config: { params: { height: pomHeight.start } } });
+
+  const blockTimestamp = useMemo(() => blocks?.data?.[0]?.timestamp || {}, [blocksAtHeight]);
+  const currentHeight = useMemo(() => blocks?.data?.[0]?.height || {}, [blocks]);
```
Comment (ikem-legend) [inline src/modules/dpos/validator/components/WarnPunishedDelegate/index.js]: @eniolam1000752 Shouldn't the fallback for timestamp or height be a number?

File: src/modules/dpos/validator/utils/getMaxAmount.js (hunk 6 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/utils/getMaxAmount.js:76]: added todo

File: src/modules/dpos/validator/utils/getMaxAmount.js (hunk 5 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/utils/getMaxAmount.js:69]: added todo

File: src/modules/dpos/validator/utils/getMaxAmount.js (hunk 4 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/utils/getMaxAmount.js:63]: added todo

File: src/modules/dpos/validator/components/EditVote/index.js (hunk 2 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/components/EditVote/index.js:59]: added comments for easy look through

File: src/modules/dpos/validator/components/DelegateSummary/DelegateSummary.js (hunk 1 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/components/DelegateSummary/DelegateSummary.js:41]: added todo

**Review by ikem-legend — APPROVED** [review]

**Review by reyraa — APPROVED** [review]

**Pull request merged**